packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -1764,6 +1774,16 @@ export default (superClass: Class<Parser>): Class<Parser> =>
         this.finishNode(node, bodilessType);
         return;
       }
+      if (bodilessType === "TSDeclareFunction" && this.state.isDeclareContext) {
+        this.raise(node.start, TSErrors.DeclareFunctionHasImplementation);
+        if (
+          // $FlowIgnore
+          node.declare
+        ) {
+          this.finishNode(node, bodilessType);
+          return;
+        }
```

Comment:
Question: Why do we need to special-case this, rather than just parsing the (invalid) function body?

Reply:
What does `special-case` mean?

Reply:
I don't know if it's the same as what you're pointing out, I noticed a my mistake. I should have used `super.parseFunctionBodyAndFinish`.

Reply:
I think you can remove this whole `if`, since `parseFunctionBodyAndFinish` would already be done anyway at the end of the function.

Reply:
I got it. But I think we need this `if` to use `bodilessType` as the function node type.

Reply:
Ohhh I missed it :upside_down_face: